src/seedu/addressbook/data/person/Address.java

```diff
@@ -11,8 +11,17 @@
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
```

Comment:
How/where are you validating if the address has all 4 components? may be the program will crash with array index out of bound exception when you try to access some component that is not present, no?